doc/proposals/ansible-operator-webhooks.md

```diff
@@ -0,0 +1,81 @@
+    are necessary for higher level validation of incoming resources, for example SAR checks. There is an open community 
+    issue explicitly requesting this feature: https://github.com/operator-framework/operator-sdk/issues/1658 .
+
+Currently Ansible-based operators have no webhook support at all, a user would have to write their own webhook server 
+and deploy it alongside the operator to get that functionality.
+
+## Goals
+- Allow the creation and use of conversion/validating/mutating webhooks implemented with Ansible
+
+## Non-Goals
+- Handle xWebhookConfiguration objects or associated Services
+- Handle cert management
+
+## Solution
+Add a webhooks section to the `watches.yaml`, that allows the user to specify lists of validating/mutating/conversion 
+webhooks that should be created. 
+
+The user should only need to provide a playbook or a role, and the url path for the webhook. 
+
+The rest of the webhook configuration, ie the creation of the Service that points to the webhook server, the Webhook 
+Configuration resources that lay out the rules and point to the proper Service, and the management of the certificates 
+for the webhook server, will be the responsiblity of the user. 
+
+The webhook configuration will look roughly as follows, with a big asterisk on the conversion webhooks which need some
+further investigation.
+
+```yaml
+---
+- version: v1alpha1
+  group: apps.example.com
+  kind: MyApp
+  role: /opt/ansible/roles/myapp
+  webhooks:
+    validating:
+    - playbook: /opt/ansible/validate.yml
+      path: /validate
+    mutating:
+    - role: /opt/ansible/roles/mutate-myapp
+      path: /mutate
+    conversion:
+    - playbook: /opt/ansible/conversion.yml
+      path: /convert
+```
+
+The justification for allowing multiple webhooks per type is that webhooks can select objects by a variety of
+attributes, so you could have a different webhook which is selected based on `metadata.labels`.
+
+Based on the webhook configuration, during initialization we will create a webhook server and add it to manager,
+and then register each of the webhooks to the webhook server, at the given path. We will create handlers that
+respond to the requests by running the associated Ansible roles or playbooks.
+
+- For validating webhooks, playbook success => Allowed, and playbook failure => Denied. We will report the
+    failureMessages in the denial response.
+- For mutating webhooks, we will create a custom module that will be used to report back the final desired
+    object. We may need to write additional logic here (either in the module or on the Go-side) that generates
+    a proper patch object from this.
+- Conversion webhooks should work roughly the same as mutating webhooks.
+
+## Discussion / Further Investigation
+- For non-trivial playbooks, will the performance be acceptable?
+- How much webhook configuration should we support?
+- Should we support side-effects?
```

Comment:
My initial guess would be no. 

There are other ways to do the side effects (in the controller), and therefore we don't restrict the use cases IMO